### PR TITLE
Update the release process

### LINKED ISSUE #249: Update the release doc follow the release template
Here is the template https://streamnative.slab.com/posts/repository-release-template-prb0lvyt

If you have any questions about the release doc template, please **leave your comment** on it, **do not edit it directly**.

### PR DESCRIPTION
**Pull request by zymap**
---

Fixes #249

*Motivation*

Update the release process.

### CODE REVIEW

**Comment by sijie** [comment]
@Huanli-Meng @Jennifer88huang Can any of you review the documentation change?

File: docs/en/release_process.md [hunk 1]
```diff
@@ -2,7 +2,19 @@
...
+
+All StreamNative repository should following the naming convention:
+
+- Branch name: `branch-X.Y.Z`
+- Tag name: `vX.Y.Z(.M)`(stable)
+- Release candidate tag: `vX.Y.Z(.M)-rc-$(date +%Y%m%d%H%M)` (unstable)
+
+`(.M)`  means our internal version release number, most of our repository is an extensions/tools for apache/pulsar. To keep track on the repository is produced by which version of the apache/pulsar, we will carry the apache/pulsar version number and using the `.M` to represent our internal release version. And all repository should keep align with streamnative/pulsar.
+
+There has two type of the tags, one is stable `vX.Y.Z(.M)`, and another is unstable `vX.Y.Z(.M)-rc-$(date +%y%m%d%H%M)`. A stable tag represent this release is a verified release, and an unstable tag represent this release is not verified.
```
Comment (Jennifer88huang) [inline docs/en/release_process.md:15]:
```suggestion
`(.M)`  is the internal version release number. Most of our repository is an extensions/tools for apache/pulsar. To keep track on the repository that is produced by which version of the apache/pulsar, we use the apache/pulsar version number and use the `.M` to represent our internal release version. All repositories should be align with streamnative/pulsar.

Two types of the tags are available, one is stable `vX.Y.Z(.M)`, and the other is unstable `vX.Y.Z(.M)-rc-$(date +%y%m%d%H%M)`. The stable tag represents that the release is a verified release, and the unstable tag represents that the release is not verified.
```

**Review by Jennifer88huang — COMMENTED** [review]
Who are the target audiences?

**Comment by zymap** [comment]
@Jennifer88huang The engineers.

**Review by Jennifer88huang — APPROVED** [review]

**Review by sijie — APPROVED** [review]

**Pull request merged**